scripts/exregional_process_bufr.sh

```diff
@@ -0,0 +1,382 @@
+#-----------------------------------------------------------------------
+#
+{ save_shell_opts; . $USHdir/preamble.sh; } > /dev/null 2>&1
+#
+#-----------------------------------------------------------------------
+#
+# Get the full path to the file in which this script/function is located 
+# (scrfunc_fp), the name of that file (scrfunc_fn), and the directory in
+# which the file is located (scrfunc_dir).
+#
+#-----------------------------------------------------------------------
+#
+scrfunc_fp=$( readlink -f "${BASH_SOURCE[0]}" )
+scrfunc_fn=$( basename "${scrfunc_fp}" )
+scrfunc_dir=$( dirname "${scrfunc_fp}" )
+#
+#-----------------------------------------------------------------------
+#
+# Print message indicating entry into script.
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+
+This is the ex-script for the task that runs bufr (cloud, metar, lightning) preprocess
+with FV3 for the specified cycle.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+START_DATE=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
+YYYYMMDDHH=$(date +%Y%m%d%H -d "${START_DATE}")
+JJJ=$(date +%j -d "${START_DATE}")
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+
+YYJJJHH=$(date +"%y%j%H" -d "${START_DATE}")
+PREYYJJJHH=$(date +"%y%j%H" -d "${START_DATE} 1 hours ago")
+
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for BUFR observation process ..."
+
+cd_vrfy ${DATA}
+
+fixgriddir=$FIXgsi/${PREDEF_GRID_NAME}
+
+print_info_msg "$VERBOSE" "fixgriddir is $fixgriddir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixgriddir}/fv3_grid_spec          fv3sar_grid_spec.nc
+
+#-----------------------------------------------------------------------
+#
+#   copy bufr table
+#
+#-----------------------------------------------------------------------
+BUFR_TABLE=${FIXgsi}/prepobs_prep_RAP.bufrtable
+cp_vrfy $BUFR_TABLE prepobs_prep.bufrtable
+
+#-----------------------------------------------------------------------
+#
+#   set observation soruce 
+#
+#-----------------------------------------------------------------------
+SUBH=""
```

Comment:
@mkavulich There used to be an RTMA code that sets `SUBH` to START_DATE but now that is gone it is always set to empty string.

Reply:
`SUBH` is used later in two places to define `obs_file` in `t${HH}{SUBH}z` although it is empty for non-RTMA cases. I could remove it completely in all places or keep the definition as empty so that it would be easy to identify which lines should change in case RTMA is added back.